atlp-rwanda/tech-strikers-bn-backend
# #175636846 logout feature

**What does this PR do?**
- User logout to end the session
 **Description of Task to be completed?**
- check if the user login in by providing a token in header
- put the token in the blacklist
- check if the token that requested resource is in the blacklist and block it
**How should this be manually tested?**
- `run git clone https://github.com/atlp-rwanda/tech-strikers-bn-backend.git`
- ` run cd tech-strikers-bn-backend`
- `run git checkout ft-logout-#1756368`
- `run npm install`
 - create your .env file from .env.example file and add variable ACCESS_TOKEN_SECRET in that file with your random value
 - add database username, password and database name in your .env file
-` run sequelize db:migrate`
- `run sequelize db:seed:all` in order to have the seeds and apply login on postman
-` run npm run dev`
- use postiman with `localhost:3000/api/V1/user/logout` to logout or end the session
- `run npm run test` to test the login API. It should pass when 
**Any background context you want to provide?**
no
 **What are the relevant pivotal tracker stories?**
- [#175636846]("https://www.pivotaltracker.com/n/projects/2475704")
**Screenshots (if appropriate)**
![image](https://user-images.githubusercontent.com/70943045/100780378-5086cf80-3412-11eb-8554-2e796701ed5e.png)
![image](https://user-images.githubusercontent.com/70943045/101465803-1012de80-3949-11eb-8f6d-ffd6b97bed17.png)

**Questions:**
